components/layout/table_cell.rs

```diff
@@ -248,21 +247,16 @@ impl Flow for TableCellFlow {
         self.block_flow.update_late_computed_block_position_if_necessary(block_position)
     }
 
-    fn build_display_list(&mut self, state: &mut DisplayListBuildState) {
-        if !self.visible {
-            return
-        }
+    fn build_display_list(&mut self, _: &mut DisplayListBuildState) {
+        use style::servo::restyle_damage::ServoRestyleDamage;
+        // This is handled by TableCellStyleInfo::build_display_list()
+        // when the containing table builds its display list
 
-        let border_painting_mode = match self.block_flow
-                                             .fragment
-                                             .style
-                                             .get_inheritedtable()
-                                             .border_collapse {
-            BorderCollapse::Separate => BorderPaintingMode::Separate,
-            BorderCollapse::Collapse => BorderPaintingMode::Collapse(&self.collapsed_borders),
-        };
-
-        self.block_flow.build_display_list_for_block(state, border_painting_mode)
+        if self.visible {
+            // we skip setting the damage in TableCellStyleInfo::build_display_list()
+            // because we only have immutable access
+            self.block_flow.fragment.restyle_damage.remove(ServoRestyleDamage::REPAINT);
```

Comment:
This should probably be unconditional.  (I think this was a bug in the old code.)